dockstore-client/src/main/java/io/dockstore/client/cli/YamlVerifyUtility.java

```diff
@@ -0,0 +1,182 @@
+/*
+ *    Copyright 2022 OICR and UCSC
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+package io.dockstore.client.cli;
+
+import java.io.File;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+import java.util.ArrayList;
+import java.util.List;
+
+import io.dockstore.common.yaml.DockstoreYaml12;
+import io.dockstore.common.yaml.DockstoreYamlHelper;
+import io.dockstore.common.yaml.Service12;
+import io.dockstore.common.yaml.YamlWorkflow;
+import org.yaml.snakeyaml.Yaml;
+
+import static io.dockstore.client.cli.ArgumentUtility.out;
+
+/*
+    GENERAL STEPS:
+    1. Verfiy that YAML syntax is valid
+    2. Verify that all fields are named correctly
+    3. Verify that all required fields are present
+    4. Verify that all files exist
+ */
+
+public final class YamlVerifyUtility {
+
+    public static final String DOCKSTOREYML = ".dockstore.yml";
+    public static final String YAML = "yaml";
+
+    public static final String VALID_DOCKSTORE_YML = " is a valid dockstore yaml file and all required files are present";
+    public static final String INVALID_DOCKSTORE_YML = "Your .dockstore.yml is invalid:\n";
+
+    public static final String CONTAINS_ERRORS = " has the following errors:\n";
+    public static final String EMPTY_FILE = " is empty";
+    public static final String FILE_DOES_NOT_EXIST = " does not exist";
+    public static final String INVALID_FILE_STRUCTURE = "Your file structure has the following errors:\n";
+
+    public static final String INVALID_YAML = " is not a valid " + YAML + "file:\n";
+    // This message is displayed when it is determined that DOCKSTOREYML is a valid yaml file,
+    // but displaying this message does NOT mean DOCKSTOREYML is a valid dockstore yaml file
+    public static final String VALID_YAML_ONLY = " is a valid " + YAML + " file";
+
+
+
+    private YamlVerifyUtility() {
+    }
+    // Determines if all the files referenced in a list of strings exist
+    private static List<String> filesExist(List<String> paths, String base) {
+        List<String> missingFiles = new ArrayList<>();
+        for (String path : paths) {
+            Path pathToFile = Paths.get(base, path);
```

Comment:
We don't officially support Windows, so you can ignore this, but I wonder if this will work on a Windows machine. I forget if Java will translate a Linux relative path to a Windows relative path